File: minegame-database/src/main/java/top/nipuru/minegame/database/processor/LoadFileHandler.java
package top.nipuru.minegame.database.processor;

import lombok.AllArgsConstructor;
import top.nipuru.minegame.common.message.database.LoadFileRequest;
import top.nipuru.minegame.common.processor.RequestDispatcher;
import top.nipuru.minegame.database.DatabaseServer;

@AllArgsConstructor
public class LoadFileHandler implements RequestDispatcher.Handler<LoadFileRequest> {

    private final DatabaseServer server;

    @Override
    public void handle(RequestDispatcher.ResponseSender responseSender, LoadFileRequest request) throws Exception {
        byte[] data = server.getFileManager().getFile(request.getFilename());
        responseSender.sendResponse(data);
    }

    @Override
    public Class<LoadFileRequest> interest() {
        return LoadFileRequest.class;
    }
}

File: minegame-database/src/main/kotlin/top/nipuru/minegame/database/processor/LoadFileHandler.kt
package top.nipuru.minegame.database.processor

import com.alipay.remoting.AsyncContext
import top.nipuru.minegame.common.message.database.LoadFileRequest
import top.nipuru.minegame.common.processor.RequestDispatcher
import top.nipuru.minegame.database.file.FileManager

class LoadFileHandler : RequestDispatcher.Handler<LoadFileRequest> {

    override fun handle(asyncCtx: AsyncContext, request: LoadFileRequest) {
        val data: ByteArray = FileManager.getFile(request.filename)
        asyncCtx.sendResponse(data)
    }

    override fun interest(): Class<LoadFileRequest> {
        return LoadFileRequest::class.java
    }
}
